### ISSUE TITLE
Raise exception on init

### ISSUE THREAD

**Issue by KharchenkoDmitriy** [mfogel/django-timezone-field#71]
I do have TimeZoneField(choices_display='WITH_GMT_OFFSET') field.
At some datetimes it breaks on init with raising pytz.exceptions.NonExistentTimeError.
Date time instance that deliver to exception:
2021-03-22 00:20:50.024855 (between 12am - 1am)
2021-03-26 00:21:06.720454 (between 12am - 1am)
2021-03-26 00:10:51.758070 (between 2am - 3am) -  this one raise exception on Asia/Amman tz
To reproduce this you can run manually next code:
```
import pytz
from datetime import datetime
buggy_time = datetime.fromisoformat('2021-03-26 00:10:51.758070')
timezones = [pytz.timezone(tz) for tz in pytz.common_timezones]  # from TimeZoneField source code
for tz in timezones:
    pytz.timezone(str(tz)).utcoffset(buggy_time)  # this one will raise exception
```
[Here](https://github.com/mfogel/django-timezone-field/blob/72bb56fbde1afdac507a77f4e65316dc7c18a595/timezone_field/choices.py#L32) is a code part that raise exception.

**Comment by KharchenkoDmitriy** [comment]
Possible solution is to use some date far from DST period instead of NOW for utcoffset calculation
```
solution_time = buggy_time.replace(day=1, month=1)
 ```

**Comment by mfogel** [comment]
Thanks for the bug report @KharchenkoDmitriy 

Are you sure you're using the most recent version of this package? This package was updated in version `4.1.2` to stop using `timezone.utcoffset()` to do the calculations in the `with_gmt_offset()` function in choices.py. The fix was in this commit: 37249c608cc.

Possible duplicate of #70

**Comment by mfogel** [comment]
Closing as duplicate of #70 and fixed in `v4.1.2`, if that's not right please reopen
